src/DateTime.cc

```diff
@@ -201,6 +201,42 @@ NsType taiToUtc(NsType nsecs) {
     return nsecs - leapNSecs;
 }
 
+/** Convert a time in nsec from any time scale to TAI
+ * \param[in] nsecs Number of nanoseconds since the epoch in TAI
+ * \param[in] scale The time scale (TAI, TT or UTC)
+ * \return Number of nanoseconds since the epoch in TAI
+ */
+long long nsecAnyToTai(long long nsecs, dafBase::DateTime::Timescale scale) {
+    if (scale == dafBase::DateTime::TAI) {
+        return nsecs;
+    } else if (scale == dafBase::DateTime::TT) {
+        return nsecs - TT_MINUS_TAI_NSECS;
+    } else if (scale == dafBase::DateTime::UTC) {
+        return utcToTai(nsecs);
+    }
```

Comment:
I've never been keen on switch, because of the drop-through behavior, though I suppose with early return it doesn't matter. How strongly do you feel about changing it?

Also, can you please say more about the compiler warnings if the enum changes? (Not that I want to rely on warnings; an error is safer, though a compiler error would be ideal).